identity/app/controllers/UpsellController.scala

```diff
@@ -47,7 +47,7 @@ class UpsellController(
     with IdentitySwitches {
 
   def confirmEmailThankYou(returnUrl: Option[String]): Action[AnyContent] = if (IdentityEnableUpsellJourneysSwitch.isSwitchedOn) csrfAddToken {
-    authenticatedActions.consentAuthWithIdapiUserAction.async { implicit request =>
+    authenticatedActions.consentAuthWithIdapiUserWithEmailValidation.async { implicit request =>
```

Comment:
We shouldn't be letting any unvalidated accounts see this page so they're not signing up to newsletters etc.